test/distributed/org/apache/cassandra/fuzz/sai/SingleNodeSAITestBase.java

```diff
@@ -104,28 +111,49 @@ public static void afterClass()
     @Before
     public void beforeEach()
     {
-        cluster.schemaChange("DROP KEYSPACE IF EXISTS harry");
-        cluster.schemaChange("CREATE KEYSPACE harry WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};");
+        cluster.schemaChange("DROP KEYSPACE IF EXISTS " + KEYSPACE);
+        cluster.schemaChange("CREATE KEYSPACE " + KEYSPACE + " WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};");
     }
 
     @Test
     public void simplifiedSaiTest()
     {
-        withRandom(rng -> basicSaiTest(rng, SchemaGenerators.trivialSchema("harry", "simplified", 1000).generate(rng)));
+        withRandom(rng -> saiTest(rng,
+                                  SchemaGenerators.trivialSchema(KEYSPACE, "simplified", 1000).generate(rng),
+                                  () -> true,
+                                  DEFAULT_REPAIR_SKIP));
+    }
+
+    @Test
+    public void indexOnlySaiTest()
+    {
+        Generator<SchemaSpec> schemaGen = schemaGenerator(false);
+        withRandom(rng -> saiTest(rng, schemaGen.generate(rng), () -> true, DEFAULT_REPAIR_SKIP));
+    }
+
+    @Test
+    public void indexOnlyNoRepairSaiTest()
+    {
+        Generator<SchemaSpec> schemaGen = schemaGenerator(true);
+        withRandom(rng -> saiTest(rng, schemaGen.generate(rng), () -> true, Integer.MAX_VALUE));
```

Comment:
correct me if i am wrong, but shouldn't this be 4 tests, not 2?

```
disable read_repair + no repair
enable read_repair + with repair

Reply:
I think what I might do here is have a couple base tests, and then randomize RR, repair frequency, etc. over multiple seeds. It's taking a while to hit issues right now.

Reply:
two things i recommend if you want to add more randomness to when you do things in the SAI Harry tests; feel free to ignore me...

1) `stateful()` - this lets you define a set of "command"s to run and each time the test runs it defines a new distribution for selecting the "command"s, so each test case will try different patterns.  
2) `accord.utils.Gens#oneOf()` - this lets you do the same, but at a `Gen` level.  You define a set of things to select from and it will create a `Gen<Gen<Thing>>`.